maat-court-data-api/src/test/java/gov/uk/courtdata/reporder/controller/RepOrderControllerTest.java

```diff
@@ -410,6 +390,46 @@ void givenValidUsn_whenFindRepOrderByUsnIsInvokedAndFindsNoRepOrder_thenCorrectR
                 .andExpect(status().isOk());
     }
 
+    @Test
+    void givenValidRepId_whenFindRepOrderStateByRepIdIsInvokedAndFindsARepOrder_thenCorrectResponseIsReturned() throws Exception {
+        RepOrderStateDTO repOrderStateDTO = TestModelDataBuilder.getPopulatedRepOrderStateDTO();
+        when(repOrderService.findRepOrderStateByRepId(MAAT_REF_VALUE)).thenReturn(repOrderStateDTO);
+
+        mvc.perform(MockMvcRequestBuilders.get(ENDPOINT_URL + "/rep-order-state/" + MAAT_REF_VALUE))
+            .andExpect(status().isOk())
+            .andExpect(content().contentType(MediaType.APPLICATION_JSON))
+            .andExpect(jsonPath("$.usn").value(USN_VALUE))
+            .andExpect(jsonPath("$.maatRef").value(MAAT_REF_VALUE))
+            .andExpect(jsonPath("$.caseId").value(CASE_ID_VALUE))
+            .andExpect(jsonPath("$.caseType").value(CASE_TYPE_VALUE))
+            .andExpect(jsonPath("$.iojResult").value(IOJ_RESULT_VALUE))
+            .andExpect(jsonPath("$.iojAssessorName.fullName").value(IOJ_ASSESSOR_FULL_NAME))
+            .andExpect(jsonPath("$.iojAssessorName.userName").value(IOJ_ASSESSOR_USERNAME))
+            .andExpect(jsonPath("$.dateAppCreated").value(DATE_APP_CREATED_VALUE))
+            .andExpect(jsonPath("$.iojReason").doesNotExist())
+            .andExpect(jsonPath("$.meansInitResult").value(MEANS_INIT_RESULT_VALUE))
+            .andExpect(jsonPath("$.meansInitStatus").value(MEANS_INIT_STATUS_VALUE))
+            .andExpect(jsonPath("$.meansFullResult").doesNotExist())
+            .andExpect(jsonPath("$.meansFullStatus").doesNotExist())
```

Comment:
Do we need a test for the state of a full means assessment?

Reply:
Good question. I'm ngl, I ripped this off of the USN endpoint tests so maybe didn't think enough about it. Let me have a look. 

I know the mapping of all fields is tested in the mapper tests and I did also test this manually so I know it works... but I will think about it and get back to you!

Reply:
So I had a look at maybe creating a few different mapping scenarios to cover each of the fields but looking at the mapping tests, this is already done there. So I'm leaning on the side of this being unnecessary duplication to do it in the controller tests as well. 

Mapper tests are here: https://github.com/ministryofjustice/laa-maat-court-data-api/blob/main/maat-court-data-api/src/test/java/gov/uk/courtdata/reporder/mapper/RepOrderMapperTest.java

Reply:
Makes sense, I agree no need to duplicate as the mapping is already tested.